pkg/topology/topology.go

```diff
@@ -52,11 +52,14 @@ type Hint struct {
 type Hints map[string]Hint
 
 func getDevicesFromVirtual(realDevPath string) (devs []string, err error) {
-	if !filepath.HasPrefix(realDevPath, "/sys/devices/virtual") {
-		return nil, fmt.Errorf("%s is not a virtual device", realDevPath)
+	relPath, err := filepath.Rel("/sys/devices/virtual", realDevPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to find relative path")
 	}
 
-	relPath, _ := filepath.Rel("/sys/devices/virtual", realDevPath)
+	if strings.HasPrefix(relPath, "..") {
+		return nil, errors.Errorf("%s is not a virtual device", realDevPath)
```

Comment:
hmm is this possible still?

Reply:
To get rid of `strings.HasPrefix()`?

I guess it's possible to just compare the first byte of `relPath` with the byte representation of "..", but would it be more readable?

Reply:
I was just wondering in which scenario `strings.HasPrefix()` is still needed because I thought `relPath, err := filepath.Rel("/sys/devices/virtual", realDevPath)`  would already error on paths like `./sys/...`.

Reply:
It would, but for e.g. `/sys/devices/not_virtual` it returns `../not_virtual` with no error.

Reply:
>  it returns `../not_virtual` with no error.

oh, I see. Thanks!